src/sysupdate/sysupdate.c

```diff
@@ -956,8 +1007,37 @@ static int verb_list(int argc, char **argv, void *userdata) {
 
         if (version)
                 return context_show_version(context, version);
-        else
+        else if (FLAGS_SET(arg_json_format_flags, SD_JSON_FORMAT_OFF))
                 return context_show_table(context);
+        else {
+                _cleanup_(sd_json_variant_unrefp) sd_json_variant *json = NULL;
+                _cleanup_strv_free_ char **versions = NULL;
+                const char *current = NULL;
+
+                for (size_t i = 0; i < context->n_update_sets; i++) {
```

Comment:
FOREACH_ARRAY

Reply:
This has already been addressed in the other PR